src/fu-polkit-agent.c

```diff
@@ -0,0 +1,219 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2011 Lennart Poettering <lennart@poettering.net>
+ * Copyright (C) 2012 Matthias Klumpp <matthias@tenstral.net>
```

Comment:
I guess this was lifted from systemd?  It feels like systemd style code.  If it was new code I guess a few things would have been differently like not using variable arguments.

Reply:
> I guess this was lifted from systemd

Yes, a long time ago (LGPLv2+ also) -- we could clean up some stuff, but I'm hesitant to change the low level UNIX bits as in all honesty I don't understand all the nuances mentioned.